shell/common/engine.cc

```diff
@@ -507,20 +507,17 @@ const std::string& Engine::GetLastEntrypointLibrary() const {
   return last_entry_point_library_;
 }
 
-// The Following commented out code connects into part 2 of the split AOT
-// feature. Left commented out until it lands:
-
-// // |RuntimeDelegate|
-// void Engine::RequestDartDeferredLibrary(intptr_t loading_unit_id) {
-//   return delegate_.RequestDartDeferredLibrary(loading_unit_id);
-// }
+// |RuntimeDelegate|
+void Engine::RequestDartDeferredLibrary(intptr_t loading_unit_id) {
+  return delegate_.RequestDartDeferredLibrary(loading_unit_id);
```

Comment:
Can this return an error (or boolean to indicate one) if the request failed. I am not comfortable swallowing a potential error here.

Reply:
The idea is that errors/failures should invoke `LoadDartDeferredLibraryFailure`. Capturing/returning any errors here may inadvertently indicate to the dev that the error handling is complete. This method is meant to inform the embedder/engine that the dart lib is requested, but makes no promises that it will or even should succeed.

I can make sure I document this.

Reply:
The full LoadDartDeferredLibraryFailure implementation is coming soon in a future PR.

Reply:
Also, this request is not blocking, as the majority of the work happens async in the downloading/extracting/opening/io of the lib. Thus, even a return value of true would not indicate success since most of the work has yet to be done.